src/middlewared/middlewared/worker.py

```diff
@@ -122,7 +125,33 @@ def main_worker(*call_args):
     return res
 
 
+def watch_parent():
+    """
+    Thread to watch for the parent pid.
+    If this process has been orphaned it means middlewared process has crashed
+    and there is nothing left to do here other than commit suicide!
+    """
+    kqueue = select.kqueue()
```

Comment:
It won't be redundant to check parent pid here, in case it's already 1.

We can also check for `ProcessLookupError` in `kqueue.control` in case parent process dies before we get control.

Reply:
It will never be `1` at that stage since that process is spawned by middlewared, if parent process dies before that it would be so rare (and would mean something else is very wrong) that I am willing to pay the price for simplicity. 

I dont understand how it could ever raise `ProcessLookupError` in this code, can you explain further?

Reply:
If parent dies between execution of `os.getppid()` and `kqueue.control`, we would get `ProcessLookupError`